fiftyone/core/runs.py

```diff
@@ -293,18 +312,77 @@ def _validate_fields_match(self, key, field_name, existing_info):
             )
 
     @classmethod
-    def list_runs(cls, samples):
+    def list_runs(cls, samples, type=None):
         """Returns the list of run keys on the given collection.
 
         Args:
             samples: a :class:`fiftyone.core.collections.SampleCollection`
+            type (None): a :class:`fiftyone.core.runs.Run` type. If provided,
+                only runs that are a subclass of this type are included
 
         Returns:
             a list of run keys
         """
         dataset_doc = samples._root_dataset._doc
         run_docs = getattr(dataset_doc, cls._runs_field())
-        return sorted(run_docs.keys())
+
+        if etau.is_str(type):
+            type = etau.get_class(type)
+
+        if type is not None:
+            keys = []
+            for key in run_docs.keys():
+                run_info = cls.get_run_info(samples, key)
+                if issubclass(run_info.config.run_cls, type):
+                    keys.append(key)
+        else:
+            keys = run_docs.keys()
```

Comment:
if you only list the keys without the class, is it possible you end up with a list like ["train","train", ...]? Also, how useful is it to know keys without the class (if the key's themselves aren't descriptive)?

Reply:
I'm not following the `["train", "train", ...]` comment. `run_docs` is a dict keyed by run keys.

wrt the second point, adding the `type` arg is a way to get keys of a particular type, which I agree might be valuable sometimes.